[BE] Store helper functions C++ for python API parity

H-Huang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #82136

Add helper functions for `store.set()`, `store.compare_set()` to accept string arguments instead of vector<uint_8> and refactored some usages internally

H-Huang committed:
[BE] Store helper functions C++ for python API parity

[ghstack-poisoned]

H-Huang committed:
Update on "[BE] Store helper functions C++ for python API parity"


Add helper functions for `store.set()`, `store.compare_set()` to accept string arguments instead of vector<uint_8> and refactored some uses

[ghstack-poisoned]

H-Huang committed:
Update on "[BE] Store helper functions C++ for python API parity"


Add helper functions for `store.set()`, `store.compare_set()` to accept string arguments instead of vector<uint_8> and refactored some usages internally

[ghstack-poisoned]

rohan-varma commented on torch/csrc/distributed/c10d/Store.cpp:
```diff
@@ -17,4 +17,24 @@ void Store::setTimeout(const std::chrono::milliseconds& timeout) {
+  return std::string(value.begin(), value.end());
+}
+
+std::string Store::get_to_str(const std::string& key) {
```
don't see where this API is used?

rohan-varma reviewed (approved):
LGTM

H-Huang committed:
Update on "[BE] Store helper functions C++ for python API parity"


Add helper functions for `store.set()`, `store.compare_set()` to accept string arguments instead of vector<uint_8> and refactored some usages internally

[ghstack-poisoned]

H-Huang commented:
@pytorchbot merge
